src/components/settings/ResetRescueKey.tsx

```diff
@@ -0,0 +1,49 @@
+import log from "loglevel";
+import { BiRegularRefresh } from "solid-icons/bi";
+
+import { useGlobalContext } from "../../context/Global";
+import { generateRescueFile } from "../../utils/rescueFile";
+
+const ResetRescueKey = () => {
+    const iconSize = 16;
+
+    const { t, setRescueFile, clearSwaps, setRescueFileBackupDone } =
+        useGlobalContext();
+
+    const handleReset = async () => {
+        const confirmText = window.prompt(t("reset_rescue_key_prompt"));
+
+        if (confirmText === null) {
+            // User clicked Cancel
+            return;
+        }
+
+        if (confirmText.toLowerCase() !== "confirm") {
+            alert(t("reset_rescue_key_invalid_confirmation"));
+            return;
+        }
+
+        try {
+            await clearSwaps();
+            const newRescueFile = generateRescueFile();
+            setRescueFile(newRescueFile);
+            setRescueFileBackupDone(false);
+            log.info("Rescue key reset successfully");
+
+            window.location.reload();
+        } catch (error) {
+            alert(t("reset_rescue_key_error", { error }));
+            log.error("Failed to reset rescue key", error);
+        }
```

Comment:
Sorry I didn't think of this before:

Reloading the WebApp doesn't seem to be necessary.
I think we could simply navigate the user to the homepage and reset these signals to get a clean swap box:
```suggestion
        try {
            await clearSwaps();
            const newRescueFile = generateRescueFile();
            setRescueFile(newRescueFile);
            setRescueFileBackupDone(false);
            setSettingsMenu(false);
            setSendAmount(BigNumber(0));
            setReceiveAmount(BigNumber(0));
            navigate("/");
            log.info("Rescue key reset successfully");
            notify("success", "Successfully generated new Rescue Key");
        } catch (error) {
            alert(t("reset_rescue_key_error", { error }));
            log.error("Failed to reset rescue key", error);
        }
```
This way we also prevent a potential 404 page if the user generates a new rescue key while in `/swap/:id` page.

I've tested it and didn't encounter any issues.

Reply:
sounds good to me, applied the suggestion, I'll fix the issues that came with it later (missing imports, not using i18n etc)